shared/actions/profile/index.js

```diff
@@ -79,15 +80,32 @@ const onClickAvatar = (_, action) => {
   }
 }
 
-const submitRevokeProof = (_, action) =>
-  RPCTypes.revokeRevokeSigsRpcPromise({sigIDQueries: [action.payload.proofId]}, Constants.waitingKey)
-    .then(() => ProfileGen.createFinishRevoking())
-    .catch((error: RPCError) => {
-      logger.warn(`Error when revoking proof ${action.payload.proofId}`, error)
-      return ProfileGen.createRevokeFinishError({
-        error: 'There was an error revoking your proof. You can click the button to try again.',
+const submitRevokeProof = (state, action) => {
+  const you = TrackerConstants.getDetails(state, state.config.username)
+  if (!you || !you.assertions) return null
+  const proof = you.assertions.find(a => a.sigID === action.payload.proofId)
+  if (!proof) return null
+
+  if (proof.type === 'pgp') {
+    return RPCTypes.revokeRevokeKeyRpcPromise({keyID: proof.kid}, Constants.waitingKey)
+      .then(() =>
+        flags.useNewRouter ? null : RouteTreeGen.createNavigateTo({parentPath: [peopleTab], path: []})
+      )
+      .catch(e => {
+        logger.info('error in dropping pgp key', e)
+        return ProfileGen.createRevokeFinishError({error: `Error in dropping Pgp Key: ${e}`})
       })
-    })
+  } else {
+    return RPCTypes.revokeRevokeSigsRpcPromise({sigIDQueries: [action.payload.proofId]}, Constants.waitingKey)
+      .then(() => ProfileGen.createFinishRevoking())
```

Comment:
`finishRevoking` causes a `navigateUp`, but you also clear modals [here](https://github.com/keybase/client/pull/16506/files#diff-dd23192500e15800dc578aa31c3a3c80R30) so you end up at the root of the people tab when revoking a non-pgp.

Reply:
this is fixed in part 2